engine/action_contact.go

```diff
@@ -213,3 +214,47 @@ func (c *Client) GetContacts(ctx context.Context, metadata *Metadata, conditions
 
 	return contacts, nil
 }
+
+func (c *Client) AcceptContact(ctx context.Context, xPubID, paymail string) error {
+
+	contact, err := getContact(ctx, paymail, xPubID, c.DefaultModelOptions()...)
+	if err != nil {
+		return err
+	}
+	if contact == nil {
+		return errors.New("contact not found")
+	}
+	if contact.Status != ContactAwaitAccept {
+		return errors.New("contact does not have status awaiting")
```

Comment:
let's include current status in this error

Reply:
included